Modules/Loadable/Markups/qSlicerMarkupsModuleWidget.cxx

```diff
@@ -2403,6 +2450,61 @@ void qSlicerMarkupsModuleWidget::onResetToDefaultDisplayPropertiesPushButtonClic
   this->markupsLogic()->SetDisplayNodeToDefaults(displayNode);
 }
 
+//-----------------------------------------------------------------------------
+void qSlicerMarkupsModuleWidget::onapplyCurveResamplingPushButtonClicked()
+{
+  Q_D(qSlicerMarkupsModuleWidget);
+
+  double resampleRate = d->SelectNumberOfSamplePointsSpinBox->value();
+
+  vtkMRMLMarkupsCurveNode *markupsCurveNode = vtkMRMLMarkupsCurveNode::SafeDownCast(d->MRMLNodeComboBox_inputNodeSelector->currentNode());
+
+  if (!markupsCurveNode)
+    {
+	vtkMRMLMarkupsClosedCurveNode *markupsCurveNode = vtkMRMLMarkupsClosedCurveNode::SafeDownCast(d->MRMLNodeComboBox_inputNodeSelector->currentNode());
+	}
+  if(markupsCurveNode)
+	{
+	vtkMRMLMarkupsCurveNode *outputNode = vtkMRMLMarkupsCurveNode::SafeDownCast(d->MRMLNodeComboBox_outputNodeSelector->currentNode());
+	if (!outputNode)
+	  {
+	  vtkMRMLMarkupsClosedCurveNode *outputNode = vtkMRMLMarkupsClosedCurveNode::SafeDownCast(d->MRMLNodeComboBox_outputNodeSelector->currentNode());
+	  if (!outputNode)
+		{
+		qDebug() << "No valid input curve. ";
+		return;
+		}
+
+	  }
+	if (outputNode->GetClassName() != markupsCurveNode->GetClassName())
```

Comment:
When the input node is selected then you can configure the output node selector to only show nodes of the same class.